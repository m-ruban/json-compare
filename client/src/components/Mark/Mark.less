.json-mark {
  color: white;
  margin-right: 3px;
  position: relative;
  top: -1px;

  &_req {
    background-color: #bba41d;
    margin-left: 3px;
  }

  &_diff {
    background-color: #ba2102;
    margin-left: 3px;
  }

  &_string {
    background-color: #28a745;
  }

  &_number {
    background-color: #989898;
  }

  &_boolean {
    background-color: #c1b58a;
  }

  &_array {
    background-color: #32c1bc;
  }

  &_object {
    background-color: #6459c1;
  }

  &_null {
    background-color: #1417ce;
  }

  &_switch {
    background-color: transparent;
    border: 1px solid #989898;
    color: #989898;
    padding: 0.23em 0.4em;
  }

  &_switch:hover {
    cursor: pointer;
  }
}
