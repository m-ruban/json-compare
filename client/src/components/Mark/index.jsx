import React from 'react';
import PropTypes from 'prop-types';
import { Badge } from 'reactstrap';
import classNames from 'classnames';
import 'components/Mark/Mark.less';

const Mark = ({ type, text, pill, onClick }) => (
    <Badge className={classNames('json-mark', `json-mark_${type}`)} color="json" pill={pill} onClick={onClick}>
        {text}
    </Badge>
);

Mark.propTypes = {
    type: PropTypes.string,
    text: PropTypes.node,
    pill: PropTypes.bool,
    onClick: PropTypes.func,
};

Mark.defaultProps = {
    pill: false,
    onClick: () => {},
};

export default Mark;
