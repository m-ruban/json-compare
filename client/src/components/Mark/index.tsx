import React from 'react';

import classNames from 'classnames';
import { Badge } from 'reactstrap';

import 'components/Mark/Mark.less';

interface IMarkProps {
    type: string;
    text: any;
    pill: boolean;
    onClick: () => void;
}

const Mark = ({ type, text, pill, onClick }: IMarkProps) => (
    <Badge className={classNames('json-mark', `json-mark_${type}`)} color="json" pill={pill} onClick={onClick}>
        {text}
    </Badge>
);

Mark.defaultProps = {
    onClick: () => null,
    pill: false,
};

export default Mark;
