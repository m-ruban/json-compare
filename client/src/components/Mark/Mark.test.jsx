import React from 'react';
import { render } from 'enzyme';
import Mark from 'components/Mark/index';

describe('<Mark />', () => {
    it('Renders correctly', () => {
        const props = { type: 'number', text: 'test', pill: false, onClick: () => {} };
        const output = render(<Mark {...props} />);
        expect(output).toMatchSnapshot();
    });
});
