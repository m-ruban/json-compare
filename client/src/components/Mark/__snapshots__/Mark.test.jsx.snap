// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mark /> Renders correctly 1`] = `
<span
  class="json-mark json-mark_number badge badge-json"
>
  test
</span>
`;
